library/core/src/cmp.rs

```diff
@@ -1481,14 +1481,11 @@ pub trait PartialOrd<Rhs: PointeeSized = Self>: PartialEq<Rhs> + PointeeSized {
     }
 }
 
-fn default_chaining_impl<T: PointeeSized, U: PointeeSized>(
+fn default_chaining_impl<T: PartialOrd<U> + PointeeSized, U: PointeeSized>(
     lhs: &T,
     rhs: &U,
     p: impl FnOnce(Ordering) -> bool,
-) -> ControlFlow<bool>
-where
-    T: PartialOrd<U>,
-{
+) -> ControlFlow<bool> {
```

Comment:
Remark: I don't find this materially better

Reply:
Please move everything into the `where` clauses instead.